File: swift/algorithms/BubbleSort.swift
import Foundation

//Implementation of BubbleSort
//****************************

func bubbleSort( _ a:inout [Int]){
    let size = a.count
    for i in 0..<size{
      let k = size-i-1
        for j in 0..<k{
            if(a[j+1]<a[j]){
                a.swapAt(j, j+1)
                }
        }
    }
    
}


//Testing the code
//****************

var a = [1,2,3,6,3,4,1,2]
bubbleSort(&a)
print(a)

var b = [1]
bubbleSort(&b)
print(b)

var c = [0,0,0,5,0,9]
bubbleSort(&c)
print(c)

File: javascript/Sorting/bubbleSort.js
function bubbleSort(values) {
  for (let i = values.length - 1; i >= 0; i--) {
    for(var j = 1; j <= i; j++) {
      if(values[j - 1] > values[j]) {
          values[j - 1] = values[j - 1] + values[j];
		      values[j] = values[j - 1] - values[j];
		      values[j - 1] = values[j - 1] - values[j];
        }
     }
   }
   return values;
}

console.log(bubbleSort([93, 2, 82, 44, 210, 1, 39])); 
